File: src/main/java/me/duncte123/skybot/commands/music/PauseCommand.java
package me.duncte123.skybot.commands.music;

import java.time.Instant;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;

import me.duncte123.skybot.Command;
import me.duncte123.skybot.SkyBot;
import me.duncte123.skybot.audio.GuildMusicManager;
import me.duncte123.skybot.utils.AudioUtils;
import me.duncte123.skybot.utils.Config;
import net.dv8tion.jda.core.EmbedBuilder;
import net.dv8tion.jda.core.entities.Guild;
import net.dv8tion.jda.core.events.message.MessageReceivedEvent;

public class PauseCommand implements Command {

    @Override
    public boolean called(String[] args, MessageReceivedEvent event) {


        if(!event.getGuild().getAudioManager().getConnectedChannel().getMembers().contains(event.getMember())){
            EmbedBuilder eb = new EmbedBuilder()
                    .setColor(Config.defaultColour)
            .addField(SkyBot.au.embedTitle, "I'm sorry, but you have to be in the same channel as me to use any music related commands", false);
            eb.setFooter(Config.defaultName, Config.defaultIcon)
            .setTimestamp(Instant.now());
            event.getTextChannel().sendMessage(eb.build()).queue();
            return false;
        }

        return true;
    }

    @Override
    public void action(String[] args, MessageReceivedEvent event) {
        AudioUtils au = SkyBot.au;

        Guild guild = event.getGuild();
        GuildMusicManager mng = au.getMusicManager(guild);
        AudioPlayer player = mng.player;

        EmbedBuilder eb = new EmbedBuilder()
                .setColor(Config.defaultColour);

        if (player.getPlayingTrack() == null){
            eb.addField(au.embedTitle, "Cannot pause or resume player because no track is loaded for playing.", false)
            .setFooter(Config.defaultName, Config.defaultIcon)
            .setTimestamp(Instant.now());
            event.getTextChannel().sendMessage(eb.build()).queue();
            return;
        }

        player.setPaused(!player.isPaused());
        if (player.isPaused()){
            eb.addField(au.embedTitle, "The player has been paused.", false);
        }else{
            eb.addField(au.embedTitle, "The player has resumed playing.", false);
        }
        eb.setFooter(Config.defaultName, Config.defaultIcon)
        .setTimestamp(Instant.now());
        event.getTextChannel().sendMessage(eb.build()).queue();
    }

    @Override
    public String help() {
        // TODO Auto-generated method stub
        return "pauses the current song";
    }

    @Override
    public void executed(boolean success, MessageReceivedEvent event) {
        // TODO Auto-generated method stub

    }

}

File: bot/src/main/kotlin/me/duncte123/skybot/commands/music/PauseCommand.kt
/*
 * Skybot, a multipurpose discord bot
 *      Copyright (C) 2017  Duncan "duncte123" Sterken & Ramid "ramidzkh" Khan & Maurice R S "Sanduhr32"
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package me.duncte123.skybot.commands.music

import me.duncte123.botcommons.messaging.MessageUtils.sendMsg
import me.duncte123.skybot.Variables
import me.duncte123.skybot.entities.jda.DunctebotGuild
import me.duncte123.skybot.objects.command.CommandContext
import me.duncte123.skybot.objects.command.MusicCommand
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class PauseCommand : MusicCommand() {
    init {
        this.name = "pause"
        this.aliases = arrayOf("resume")
        this.help = "Pauses the current song"
    }

    override fun run(ctx: CommandContext) {
        val mng = ctx.audioUtils.getMusicManager(ctx.guildId)
        val localPlayer = mng.player

        localPlayer.ifPresentOrElse({ player ->
            if (player.track == null) {
                sendMsg(ctx, "Cannot pause or resume player because no track is loaded for playing.")
                return@ifPresentOrElse
            }

            player.setPaused(!player.paused).subscribe {
                sendMsg(ctx, "The player has ${if (it.paused) "been paused" else "resumed playing"}.")
            }
        }) {
            sendMsg(ctx, "Cannot pause or resume player because no track is loaded for playing.")
        }
    }

    override fun handleEvent(
        event: SlashCommandInteractionEvent,
        guild: DunctebotGuild,
        variables: Variables,
    ) {
        val mng = variables.audioUtils.getMusicManager(event.guild!!.idLong)
        val localPlayer = mng.player

        localPlayer.ifPresentOrElse({ player ->
            if (player.track == null) {
                event.reply("Cannot pause or resume player because no track is loaded for playing.").queue()
                return@ifPresentOrElse
            }

            player.setPaused(!player.paused).subscribe {
                event.reply("The player has ${if (it.paused) "been paused" else "resumed playing"}.").queue()
            }
        }) {
            event.reply("Cannot pause or resume player because no track is loaded for playing.").queue()
        }
    }
}
